pkg/publisher/publisher.go

```diff
@@ -126,14 +98,22 @@ func New(ctx context.Context) (Publisher, error) {
 	}
 	clusterID := getClusterID(ec2Client)
 
-	// Get CloudWatch client
-	ec2MetadataClient := ec2metadatawrapper.New(nil)
+	// Get ec2metadata client
+	ec2MetadataClient := ec2metadatawrapper.New(sess)
 
 	region, err := ec2MetadataClient.Region()
 	if err != nil {
 		return nil, errors.Wrap(err, "publisher: Unable to obtain region")
 	}
-	cloudwatchClient := cloudwatch.New(awsSession, aws.NewConfig().WithMaxRetries(cloudwatchClientMaxRetries).WithRegion(region))
+
+	// Get AWS session
+	awsCfg := aws.Config{
+		Region: aws.String(region),
+	}
+	sess = sess.Copy(&awsCfg)
```

Comment:
Same comment as before, if we can pass the region to "sess", then we can avoid a copy.